### ISSUE TITLE
Custom apps for workers are not working

### ISSUE THREAD

**Issue by nukah** [celery/celery#323]
With new low-level parameters for custom apps/loaders/config there comes an issue.
Basicly starting new worker is going through WorkerCommand -> handle_args -> parse_args -> fill configuration with new parameters and default parameters and start the app.

Problem is, when you start up the worker, it will call for argument_parser first, that's where it loads information about your custom app, then it goes for parsing and compiling full option list. And here comes the main issue - it will call for get_options, which is getting worker-specific-options and if they are not set in command-line, it defaults them to the content of self.app.conf, which itself has not yet been instantiated. 

In short, with --app='module' , Command.**init**(app) will always be None, and default app(celery.app.App) will start no matter what option is set.
